triqs/utility/timestamp.hpp

```diff
@@ -0,0 +1,54 @@
+
+/*******************************************************************************
+ *
+ * TRIQS: a Toolbox for Research in Interacting Quantum Systems
+ *
+ * Copyright (C) 2011 by Hugo U.R. Strand
+ *
+ * TRIQS is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * TRIQS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * TRIQS. If not, see <http://www.gnu.org/licenses/>.
+ *
+ ******************************************************************************/
+#pragma once
+
+#include <iomanip>
+#include <chrono>
+#include <sstream>
+
+namespace triqs {
+  namespace utility {
+
+    std::string inline timestamp() {
+      std::ostringstream s;
+      auto now          = std::chrono::system_clock::now();
+      std::time_t now_c = std::chrono::system_clock::to_time_t(now);
+      s << std::put_time(std::localtime(&now_c), "%H:%M:%S");
+      return s.str();
+    }
+
+    std::string inline hours_minutes_seconds_from_seconds(double seconds) {
+      using namespace std::chrono;
+      time_point<system_clock> tp(milliseconds(long(seconds * 1000)) - hours(1));
+      std::ostringstream s;
+      std::time_t tp_c = system_clock::to_time_t(tp);
+      s << std::put_time(std::localtime(&tp_c), "%H:%M:%S");
+      return s.str();
+    }
+    
+    std::string inline estimate_time_left(int Ntot, int iter, timer & Timer) {
```

Comment:
Fair comment.

I was trying to make the additions to `mc_generic.hpp` as simple as possible, and factoring out this logic makes it easier to read that code. If we where to incorporate this code in the `mc_generic` run method there would be rather verbose call to `hours_minutes_seconds_from_seconds(...)` in that method. I argue that it is easier to read `estimate_time_left(...)`, which tells you exactly what is going on. It does not even require a comment on what is happening since the function name is self-explanatory.